src/lib/computing.ts

```diff
@@ -0,0 +1,116 @@
+import { accounts, program, post } from 'aleph-sdk-ts';
+
+import { DEFAULT_API_V2 } from 'aleph-sdk-ts/global';
+import { ItemType } from 'aleph-sdk-ts/messages/message';
+import { ALEPH_CHANNEL } from 'config/constants';
+
+import { IPCProgram, ResponseType } from 'types/types';
+
+class Computing {
+	public programs: IPCProgram[];
+
+	private readonly account: accounts.base.Account | undefined;
+
+	constructor(importedAccount: accounts.base.Account) {
+		this.programs = [];
+		this.account = importedAccount;
+	}
+
+	public async loadPrograms(): Promise<ResponseType> {
+		try {
+			if (this.account) {
+				const userData = await post.Get({
+					APIServer: DEFAULT_API_V2,
+					types: '',
+					pagination: 200,
+					page: 1,
+					refs: [],
+					addresses: [this.account.address],
+					tags: [],
+					hashes: [],
+				});
+
+				userData.posts.map(async (postContent) => {
+					const itemContent = JSON.parse(postContent.item_content);
+
+					if (itemContent.content.headers === 'InterPlanetaryCloud2.0 - Programs') {
+						console.log('Post program founded');
+
+						const foundedProgram: IPCProgram = {
+							hash: itemContent.content.program.hash,
+							name: itemContent.content.program.name,
+							created_at: itemContent.content.program.created_at,
+						};
+
+						this.programs.push(foundedProgram);
+
+						return true;
+					}
+					return false;
+				});
+
+				return { success: true, message: 'Programs loaded' };
+			}
+			return { success: false, message: 'Failed to load account' };
+		} catch (error) {
+			console.error(error);
+			return { success: false, message: 'Failed to load programs' };
+		}
+	}
+
+	public async uploadProgram(myProgram: IPCProgram, uploadFile: File): Promise<ResponseType> {
+		try {
+			if (this.account) {
+				const programHashPublishProgram = await program.publish({
+					channel: ALEPH_CHANNEL,
+					account: this.account,
+					storageEngine: ItemType.storage,
+					inlineRequested: true,
+					APIServer: DEFAULT_API_V2,
+					file: uploadFile,
+					entrypoint: 'main:app',
+				});
+
+				const newProgram: IPCProgram = {
+					name: myProgram.name,
+					hash: programHashPublishProgram.item_hash,
+					created_at: myProgram.created_at,
+				};
+
+				this.programs.push(newProgram);
+
+				return { success: true, message: 'Program uploaded' };
+			}
+			return { success: false, message: 'Failed to load account' };
+		} catch (err) {
+			console.log(err);
+			return { success: false, message: 'Failed to upload program' };
+		}
+	}
+
+	public async addToUser(): Promise<ResponseType> {
+		try {
+			if (this.account) {
+				await post.Publish({
+					APIServer: DEFAULT_API_V2,
+					channel: ALEPH_CHANNEL,
+					inlineRequested: true,
+					storageEngine: ItemType.ipfs,
+					account: this.account!,
+					postType: '',
+					content: {
+						headers: 'InterPlanetaryCloud2.0 - Programs',
+						program: this.programs[this.programs.length - 1],
```

Comment:
:warning: BIG problem, for each programs uploaded you publish a new `post` message.

Why not just use one `post` message with all the programs like for the files?

Reply:
> warning BIG problem, for each programs uploaded you publish a new `post` message.
> 
> Why not just use one `post` message with all the programs like for the files?

If for each new program added to IPC I make a post message where in content I put the entire array of programs already pushed to IPC and I want to load them I just retrieve the last post with all programs of the user it would be better ?

Reply:
> If for each new program added to IPC I make a post message where in content I put the entire array of programs already pushed to IPC and I want to load them I just retrieve the last post with all programs of the user it would be better ?

You can `amend` your post to change the array.

Reply:
> > If for each new program added to IPC I make a post message where in content I put the entire array of programs already pushed to IPC and I want to load them I just retrieve the last post with all programs of the user it would be better ?
> 
> You can `amend` your post to change the array.

Even for the first post ?

Reply:
> Even for the first post ?

No you need to publish a post first and then you can amend it if the user publish a new program (just don't publish a new post for each new program, a post for each user is enough)

Reply:
Ok, I see how it works. I just did it ! :+1: